backend/tests/conftest.py

```diff
@@ -66,6 +72,33 @@ def apply_migrations() -> None:
     upgrade(config, "head")
 
 
+@pytest.fixture(scope="function", autouse=True)
+def patch_posthog(monkeypatch: pytest.MonkeyPatch) -> None:
+    class PosthogAnalyticsMock:
+        async def __aenter__(self) -> tuple[PosthogClient, UserModel]:
+            return posthog, UserModel(
+                id=uuid.uuid4(),
+                name="",
+                langsmith_api_key="",
+                sentry_enabled=False,
+                analytics_enabled=False,
+                preferred_openai_model="",
+                openai_api_key="",
+                openai_base_url="",
+            )
+
+        async def __aexit__(self, exc_type: Exception, exc_val: Exception, exc_tb: Exception) -> None:
+            pass
+
+    monkeypatch.setattr("dataline.main.PosthogAnalytics", PosthogAnalyticsMock)
+    monkeypatch.setattr("dataline.api.auth.router.PosthogAnalytics", PosthogAnalyticsMock)
+    monkeypatch.setattr("dataline.api.connection.router.PosthogAnalytics", PosthogAnalyticsMock)
+    monkeypatch.setattr("dataline.api.conversation.router.PosthogAnalytics", PosthogAnalyticsMock)
+    monkeypatch.setattr("dataline.api.result.router.PosthogAnalytics", PosthogAnalyticsMock)
+    monkeypatch.setattr("dataline.api.settings.router.PosthogAnalytics", PosthogAnalyticsMock)
+    monkeypatch.setattr("dataline.services.conversation.PosthogAnalytics", PosthogAnalyticsMock)
+
+
```

Comment:
That's annoying...

Reply:
Yupp, mocking do be like this. 

Other solution is to not mock, and instead build a dummy User + dummy client and do more stuff. This was much faster. And if you forget to include this in the future, tests will fail, so it's not really a risk.